main.py

```diff
@@ -1,410 +1,69 @@
-ground_x = 0
-ground_width = ground.get_width() - 5	# To prevent glitches in background movement...yet to find an optimal solution
-foreground_speed = 6 	# Foreground shifts by 6 pixels in each game loop
-
-# Coin collection board
-coin_board1 = pygame.image.load(os.path.join('Utils/Pics/Display','coin_display.png'))
-coin_board = pygame.transform.scale(coin_board1, (int(coin_board1.get_width()//1.5), int(coin_board1.get_height()//1.5)))
-coins = []
-num_coins_collected = 0
-
-speed = 60			# fps
-max_y_coord = 560	# player's y axis limit
-run = True
-
-obstacle_classes = [Tree, Other_obstacles]
-
-collision_occured = False
-rock_collision_occured = False
-
-player = Player(250, 313)	# Creating an instance of the class Player
-
-clock = pygame.time.Clock()
-
-# Setting a userevent once in every 1.5 seconds to generate coin
-pygame.time.set_timer(pygame.USEREVENT+1, 1500)
-
-# Setting a userevent once in every 8 seconds to generate tree obstacles
-pygame.time.set_timer(pygame.USEREVENT+2, 8000)
-
-# Setting a userevent once in every 20 seconds to generate rock obstacles
-pygame.time.set_timer(pygame.USEREVENT+3, 20000)
-
-'''
-frame_count = 0	# chck fps
-start_time = time.time()'''
-
-# GAME LOOP
-while run:
-	# Draws stuff to be displayed in window
-	bg_x, bg_width, ground_x, ground_width = draw_scene_and_obstacles(bg_x, bg_width, ground_x, ground_width)
-	
-	# Get mouse pointer coordinates
-	(mx, my) = pygame.mouse.get_pos()
-
-	# Display mouse pointer coordinates for reference
-	display_mouse_pointer_coordinates(mx,my)
-	   	
-	# limit player's movable region
-	if my < 560 :
-		player.x, player.y = 250, my
-		player.draw(win)
-	else:
-		player.x, player.y = 250, max_y_coord
-		player.draw(win)
-	
-	# Event loop
-	for event in pygame.event.get():
-		
-		if event.type == pygame.QUIT:
-			pygame.quit()
-			sys.exit()
-			run = False
-		
-		if event.type == pygame.KEYDOWN:
-			if event.key == 27:		# press esc to quit
-				pygame.quit()
-				sys.exit()
-				run = False
-		
-		if event.type == pygame.USEREVENT+1:
-			create_coin()
-		if event.type == pygame.USEREVENT+2:
-			create_tree_obstacle()
-		if event.type == pygame.USEREVENT+3:
-			create_other_obstacle()
-	
-	update_obstacle_position()
-	update_coins_position()
-
-	# Coin collection
-	collected = coin_collection()	# Checks collision and returns bool 
-	if collected:
-		num_coins_collected += 1
-	display_num_coins_collected()
+def draw_all_objects():
+	"""
+	Draws the background, foreground, obstacles and coins.
+	"""
+	bg_module.draw_bg(win)
+	obstacles_module.draw_obstacles(win)
+	coins_module.draw_coins(win)
+	fg_module.draw_fg(win)
+	player_module.draw_player(win)
 
-	# Collision with Obstacles
-	collision_occured = collision_with_obstacle()	# Checks collision and returns bool 
-	if collision_occured:		# Dummy exit
-		display_collision_message()
-		pygame.display.update()
-		time.sleep(3)
-		break
+# MAIN ALGORITHM
+if __name__ == '__main__':
+	import player_module

```

Comment:
This right here is an example of another bad programming practice! Import files should only be on the top of each file and should never play a part inside functions!

Reply:
I had to do that in order to convert the pixel format of the images while loading them. The conversion can be done only if the game window is initialized. So importing the files at the top threw errors.
Now I have added a `change_img_pixel_format` function to do that conversion after initializing the game window.